test/e2e/antreapolicy_test.go

```diff
@@ -2359,6 +2359,54 @@ func testACNPNamespaceIsolation(t *testing.T, data *TestData) {
 	executeTestsWithData(t, testCase, data)
 }
 
+func testACNPStrictNamespacesIsolation(t *testing.T) {
+	builder := &ClusterNetworkPolicySpecBuilder{}
+	builder = builder.SetName("test-acnp-strict-ns-isolation").
+		SetTier("securityops").
+		SetPriority(1.0).
+		SetAppliedToGroup([]ACNPAppliedToSpec{{NSSelector: map[string]string{}}})
+	builder.AddIngress(v1.ProtocolTCP, nil, nil, nil, nil, nil, nil, nil, nil,
+		true, nil, crdv1alpha1.RuleActionPass, "", "")
+	builder.AddIngress(v1.ProtocolTCP, nil, nil, nil, nil, nil, map[string]string{}, nil, nil,
+		false, nil, crdv1alpha1.RuleActionDrop, "", "")
+	// deny ingress traffic except from own namespace, which is delegated to Namespace owners (who can create K8s
+	// NetworkPolicies to regulate intra-Namespace traffic)
+	reachability := NewReachability(allPods, Dropped)
+	reachability.ExpectAllSelfNamespace(Connected)
+	testStep1 := &TestStep{
+		"Namespace isolation, Port 80",
+		reachability,
+		[]metav1.Object{builder.Get()},
+		[]int32{80},
+		v1.ProtocolTCP,
+		0,
+		nil,
+	}
+
+	// Add a K8s namespaced NetworkPolicy in ns x that isolates all Pods in that namespace.
+	builder2 := &NetworkPolicySpecBuilder{}
+	builder2 = builder2.SetName("x", "default-deny-in-namespace-x")
+	builder2.SetTypeIngress()
+	reachability2 := NewReachability(allPods, Dropped)
+	reachability2.ExpectAllSelfNamespace(Connected)
+	reachability2.ExpectSelfNamespace("x", Dropped)
+	reachability2.ExpectSelf(allPods, Connected)
+	testStep2 := &TestStep{
+		"Namespace isolation with K8s NP, Port 80",
+		reachability2,
+		[]metav1.Object{builder2.Get()},
+		[]int32{80},
+		v1.ProtocolTCP,
+		0,
+		nil,
+	}
+
+	testCase := []*TestCase{
+		{"ACNP strict Namespace isolation for all namespaces", []*TestStep{testStep1, testStep2}},
+	}
+	executeTests(t, testCase)
```

Comment:
Wondering if we want `executeTestsWithData` here? 🤔

Reply:
We don't. In fact, any tests that does not have custom probes won't need to be executed with data. Fixed such occurrences as well

Reply:
Got it, thanks!